modules/gdscript/gdscript_vm.cpp

```diff
@@ -679,11 +678,40 @@ Variant GDScriptFunction::call(GDScriptInstance *p_instance, const Variant **p_a
 	bool exit_ok = false;
 	bool awaited = false;
 #endif
+
 #ifdef DEBUG_ENABLED
-	int variant_address_limits[ADDR_TYPE_MAX] = { _stack_size, _constant_count, p_instance ? p_instance->members.size() : 0, script->static_variables.size() };
+	int variant_addresses_size = ADDR_TYPE_STATIC_VAR + script->inheritance_level + 1;
+
+	int *variant_address_limits = memnew_arr(int, ADDR_TYPE_STATIC_VAR + script->inheritance_level + 1);
+	variant_address_limits[ADDR_TYPE_STACK] = _stack_size;
+	variant_address_limits[ADDR_TYPE_CONSTANT] = _constant_count;
+	variant_address_limits[ADDR_TYPE_MEMBER] = p_instance ? p_instance->members.size() : 0;
 #endif
 
-	Variant *variant_addresses[ADDR_TYPE_MAX] = { stack, _constants_ptr, p_instance ? p_instance->members.ptrw() : nullptr, script->static_variables.ptrw() };
+	Variant **variant_addresses = memnew_arr(Variant *, ADDR_TYPE_STATIC_VAR + script->inheritance_level + 1);
```

Comment:
I'm not sure about the preferred way to allocate memory, maybe `new[]` or `alloca()` is better here?

Reply:
I think this is the preferred way in the Godot codebase:
https://github.com/godotengine/godot/blob/49243a9a9816b334002f7d73d9e940d00b49158d/scene/animation/tween.cpp#L748